src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,44 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.exception.TaskIndexOutOfBoundException;
+import duke.storage.Storage;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/**
+ * A class represents a DeleteCommand.
+ */
+public class DeleteCommand extends IndexCommand {
+    /**
+     * Constructs a DeleteCommand.
+     * @param fullCommand Full command from the user's input.
+     */
+    public DeleteCommand(String fullCommand) {
+        super(fullCommand);
+    }
+
+    /**
+     * Deletes the specified task from the task list.
+     * @param tasks Task list given.
+     * @param ui User interface class object.
+     * @param storage Storage path that is going to be updated.
+     * @throws DukeException If error occurs during the process.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
```

Comment:
This line here states that `DukeException` is thrown, but only `TaskIndexOutOfBoundException` is actually thrown.
This will require the stack to explicitly handle the general exception instead of a more specific one.
Consider changing it to the `throws TaskIndexOutOfBoundException`?

Same goes to the other commands.

Reply:
> This line here states that `DukeException` is thrown, but only `TaskIndexOutOfBoundException` is actually thrown.
> This will require the stack to explicitly handle the general exception instead of a more specific one.
> Consider changing it to the `throws TaskIndexOutOfBoundException`?
> 
> Same goes to the other commands.

Thanks for pointing it out! That is indeed a very good suggestion.